packages/mobile/src/paymentRequest/PaymentRequestConfirmation.tsx

```diff
@@ -23,86 +22,69 @@ import { StackParamList } from 'src/navigator/types'
 import { getDisplayName } from 'src/recipients/recipient'
-import { RootState } from 'src/redux/reducers'
-import { ConfirmationInput, getConfirmationInput } from 'src/send/utils'
+import useSelector from 'src/redux/useSelector'
+import { getConfirmationInput } from 'src/send/utils'
 import DisconnectBanner from 'src/shared/DisconnectBanner'
 import Logger from 'src/utils/Logger'
-import { currentAccountSelector } from 'src/web3/selectors'
+import { walletAddressSelector } from 'src/web3/selectors'
 
 // @ts-ignore
 const TAG = 'paymentRequest/confirmation'
 
-interface StateProps {
-  e164PhoneNumber: string | null
-  account: string | null
-  confirmationInput: ConfirmationInput
-  addressJustValidated?: boolean
-}
+type OwnProps = StackScreenProps<StackParamList, Screens.PaymentRequestConfirmation>
 
-interface DispatchProps {
-  showError: typeof showError
-  writePaymentRequest: typeof writePaymentRequest
-}
+type Props = WithTranslation & OwnProps
+
+export const paymentConfirmationScreenNavOptions = () => ({
+  ...emptyHeader,
+  headerLeft: () => <BackButton eventName={RequestEvents.request_confirm_back} />,
+})
 
-const mapDispatchToProps = { showError, writePaymentRequest }
+function PaymentRequestConfirmation({ route }: Props) {
+  const [comment, setComment] = useState('')
+  const [transactionData] = useState(route.params.transactionData)
+  const [addressJustValidated] = useState(route.params.addressJustValidated)
+
+  const e164NumberToAddress = useSelector((state) => state.identity.e164NumberToAddress)
+  const secureSendPhoneNumberMapping = useSelector(
+    (state) => state.identity.secureSendPhoneNumberMapping
+  )
+  const account = useSelector(walletAddressSelector)
 
-const mapStateToProps = (state: RootState, ownProps: OwnProps): StateProps => {
-  const { route } = ownProps
-  const { transactionData, addressJustValidated } = route.params
-  const { e164NumberToAddress } = state.identity
-  const { secureSendPhoneNumberMapping } = state.identity
   const confirmationInput = getConfirmationInput(
     transactionData,
     e164NumberToAddress,
     secureSendPhoneNumberMapping
   )
-  return {
-    confirmationInput,
-    e164PhoneNumber: state.account.e164PhoneNumber,
-    account: currentAccountSelector(state),
-    addressJustValidated,
-  }
-}
-
-type OwnProps = StackScreenProps<StackParamList, Screens.PaymentRequestConfirmation>
-
-type Props = DispatchProps & StateProps & WithTranslation & OwnProps
-
-export const paymentConfirmationScreenNavOptions = () => ({
-  ...emptyHeader,
-  headerLeft: () => <BackButton eventName={RequestEvents.request_confirm_back} />,
-})
 
-class PaymentRequestConfirmation extends React.Component<Props> {
-  state = {
-    comment: '',
+  const amount = {
+    value: confirmationInput.amount,
+    currencyCode: confirmationInput.currency,
   }
 
-  componentDidMount() {
-    const { addressJustValidated, t } = this.props
+  const { t } = useTranslation()
+
+  useEffect(() => {
```

Comment:
Lets not change this in this PR since we're not doing logic changes, but it's really weird to me that we're passing this `addressJustValidated` value here just to show a toast. We can probably do the `Logger.showMessage` in the callers side and not have the prop here. 

Just writing this to add a mental note, no need to do it now.